master/buildbot/test/integration/test_integration_secrets_with_vault.py

```diff
@@ -0,0 +1,93 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+from unittest.case import SkipTest
+
+from twisted.internet import defer
+
+from buildbot.process.properties import Interpolate
+from buildbot.secrets.providers.vault import HashiCorpVaultSecretProvider
+from buildbot.steps.shell import ShellCommand
+from buildbot.test.util.integration import RunMasterBase
+
+
+# This integration test creates a master and worker environment,
+# with one builders and a shellcommand step
+# To be run, the test use a Vault server instance.
+# To easily run a Vault instance find the dockerfile dedictaed to Vault.
+# command: docker-compose up
+# Note the Vault docker image address and port.
+# connect to the docker instance:
+# command: docker exec -i -t ``docker_vault_image_name`` /bin/sh
```

Comment:
which image name? how to build it. This would be best to have one that is already exists on docker hub